File: benchmarks/fixture/ruby/session.rb
module FixtureSession
  def self.renew_session_credentials(refresh_token, now)
    raise ArgumentError, 'refresh token required' if refresh_token.to_s.empty?

    { access_token: "access:#{refresh_token}", expires_at: now + 900 }
  end

  def self.validate_access_token(access_token, now)
    access_token[:expires_at].is_a?(Integer) && access_token[:expires_at] > now
  end
end

File: benchmarks/fixture/cpp/session.cpp
#include <stdexcept>
#include <string>

struct Credentials {
    std::string access_token;
    long expires_at;
};

Credentials renew_session_credentials(const std::string& refresh_token, long now) {
    if (refresh_token.empty()) throw std::invalid_argument("refresh token required");
    return {"access:" + refresh_token, now + 900};
}

bool validate_access_token(long expires_at, long now) {
    return expires_at > now;
}
